src/app/libraries/http_lib.php

```diff
@@ -1,14 +1,19 @@
 <?php
 
 class http_lib extends Library {
-
     function response_code($code, $view_and_exit = true, $view_name = false) {
         http_response_code($code);
         if ($view_and_exit) {
             if (empty($view_name)) {
                 $view_name = strval($code);
             }
-            $this->load_view($view_name);
+            // Check if a dedicated view exists; if not, load common error view
+            $dedicatedView = $this->load_view($code);
+            if (! $dedicatedView) {
+                $this->load_view('http_error', [
```

Comment:
That doesn't make any sense. Let's even say we do call this with 200. Why in the world would we want to load view and exit? That will only happen in the case of errors, considering we're returning an http code and then exiting.

We'll modify the code to be different for `4xx` and `5xx` when we have different error views for them, though I have no idea why that would happen. Until then, there's really no point.